src/app/games/diam/tests/diam.component.spec.ts

```diff
@@ -0,0 +1,213 @@
+    const A2: DiamPiece = DiamPiece.ZERO_SECOND;
+    const B1: DiamPiece = DiamPiece.ONE_FIRST;
+    const B2: DiamPiece = DiamPiece.ONE_SECOND;
+
+    let testUtils: ComponentTestUtils<DiamComponent>;
+
+    beforeEach(fakeAsync(async() => {
+        testUtils = await ComponentTestUtils.forGame<DiamComponent>('Diam');
+    }));
+    it('should create', () => {
+        expect(testUtils.wrapper).withContext('should be created').toBeTruthy();
+        expect(testUtils.getComponent()).withContext('Component should be created').toBeTruthy();
+    });
+    it('should allow simple drops by clicking the piece and then the target', fakeAsync(async() => {
+        // given the initial state
+
+        // when clicking on a piece and then on a space
+        // then the move is made
+        await testUtils.expectClickSuccess('#piece_0_1');
+        const move: DiamMove = new DiamMoveDrop(2, DiamPiece.ZERO_SECOND);
+        await testUtils.expectMoveSuccess('#click_2', move);
+    }));
+    it('should forbid selecting a piece of the opponent', fakeAsync(async() => {
+        // given the initial state
+        // when clicking on a piece of the opponent
+        // then the corresponding error is shown
+        await testUtils.expectClickFailure('#piece_1_0', RulesFailure.MUST_CHOOSE_PLAYER_PIECE());
+    }));
+    it('should consider a piece in game click on an opponent piece as a space click if a piece is selected for better usability', fakeAsync(async() => {
+        // given a state where there are already pieces in game
+        const state: DiamState = DiamState.fromRepresentation([
+            [__, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+            [B1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+        ], 0);
+        testUtils.setupState(state);
+        // when clicking on a remaining piece and then on a opponent piece in game
+        await testUtils.expectClickSuccess('#piece_0_0');
+        const move: DiamMove = new DiamMoveDrop(0, DiamPiece.ZERO_FIRST);
+        // then the move is made to the corresponding space
+        await testUtils.expectMoveSuccess('#click_0_1', move);
+    }));
+    it('should consider a piece in game click on a player piece as a regular piece click', fakeAsync(async() => {
+        // given a state where there are already pieces in game
+        const state: DiamState = DiamState.fromRepresentation([
+            [__, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+            [B1, __, __, __, __, __, __, __],
+            [A1, __, __, __, __, __, __, __],
+        ], 0);
+        testUtils.setupState(state);
+        // when clicking on a remaining piece and then on a player piece in game
+        await testUtils.expectClickSuccess('#piece_0_0');
+        await testUtils.expectClickSuccess('#click_0_0');
+        // then no move is made and the new piece is selected
```

Comment:
// given a state where there are already piece on board and a remainingPiece is selected
// when cliccking on a piece on board
// then no move should be done and piece should be selected

Surtout ne pas oublier de laisser la type de ma deuxième ligne, c't'important!

Reply:
yep c'est mieux ainsi, fait